patch.crates-io.sh

```diff
@@ -88,11 +88,14 @@ done
 echo "Patching in $solana_ver from $solana_dir"
 echo
 for crate in "${workspace_crates[@]}"; do
-  if grep -q '\[patch.crates-io\]' "$crate"; then
+  if grep -q "# The following entries are auto-generated by $0" "$crate"; then
     echo "$crate is already patched"
   else
+    if ! grep -q '\[patch.crates-io\]' "$crate"; then
```

Comment:
i know this is super inefficient for invoking `grep` many time but this is simplest solution and who cares for this tiny script? :)